pkg/workceptor/remote_work.go

```diff
@@ -97,19 +119,21 @@ func (rw *remoteUnit) connectAndRun(ctx context.Context, action actionFunc) erro
 // getConnectionAndRun retries connecting to a host and, once the connection succeeds, runs an action function.
 // If firstTimeSync is true, a single attempt is made on the calling goroutine. If the initial attempt fails to
 // connect or firstTimeSync is false, we run return ErrPending to the caller.
-func (rw *remoteUnit) getConnectionAndRun(ctx context.Context, firstTimeSync bool, action actionFunc) error {
+func (rw *remoteUnit) getConnectionAndRun(mw *utils.JobContext, firstTimeSync bool, action actionFunc) error {
 	if firstTimeSync {
-		err := rw.connectAndRun(ctx, action)
+		err := rw.connectAndRun(mw, action)
 		if err == nil {
 			return nil
 		} else if !utils.ErrorIsKind(err, "connection") {
 			return err
 		}
 	}
 	go func() {
-		conn, reader := rw.getConnection(ctx)
+		conn, reader := rw.getConnection(mw)
 		if conn != nil {
-			_ = action(ctx, conn, reader)
+			_ = action(mw, conn, reader)
+		} else {
+			mw.WorkerDone()
```

Comment:
action() already calls WorkerDone() in the right places, so we need to call WorkerDone() only if action() isn't executed

Reply:
This can't be right ... action is of type actionFunc:
```
type actionFunc func(context.Context, net.Conn, *bufio.Reader) error
```
which takes a Context, not a JobContext, so it doesn't have a WorkerDone function to call.

Reply:
it appears it grabs the mw out of scope?

https://github.com/project-receptor/receptor/blob/e862c28921c3edd780ffbcfb2a0fd0dccac5e09c/pkg/workceptor/remote_work.go#L407

Reply:
That's being called from the outer function, not the action passed in.